### ISSUE TITLE
merkaatror segfaults when opening mdc file (in attachment)

### ISSUE THREAD

**Issue by tomhughes** [tomhughes/trac-tickets#1546]
**[Submitted to the original trac issue database at 9.13am, Saturday, 31st January 2009]**

Hi! I have Fedora 10 x86_64 & merkaartor 0.13svn-v13450, when I try to open file Soblahov.mdc (in attachment) merkaartor segfaults. Logs from dmesg: merkaartor[3627]: segfault at 0 ip 0000000000427bb2 sp 00007fff6ec3f900 error 4 in merkaartor[400000+355000]. Another mdc files are OK.

**Comment by tomhughes** [comment]
**Author: fuuzuu**
**[Added to the original trac issue at 9.16am, Saturday, 31st January 2009]**

cannot upload file - it is too big (1MB)

**Comment by tomhughes** [comment]
**Author: fuuzuu**
**[Added to the original trac issue at 9.19am, Saturday, 31st January 2009]**

uploaded here: http://rapidshare.com/files/191935536/Soblahov.mdc.html

**Comment by tomhughes** [comment]
**Author: Koying**
**[Added to the original trac issue at 10.15am, Saturday, 31st January 2009]**

If you want to load your file, just change in line 8909


```
  <CommandHistory index="534" >
```

by 


```
  <CommandHistory index="500" >
```

Now, I'll look into why it happened...

**Comment by tomhughes** [comment]
**Author: Koying**
**[Added to the original trac issue at 10.21am, Saturday, 31st January 2009]**

Hold on, the 500 should be **533**

**Comment by tomhughes** [comment]
**Author: fuuzuu**
**[Added to the original trac issue at 10.49am, Saturday, 31st January 2009]**

Replying to [comment:4 Koying]:
> Hold on, the 500 should be **533**
Thank you - it's working, one more question: when I close it without any changes, the "Unsaved changes" dialog is shown - why?

**Comment by tomhughes** [comment]
**Author: Koying**
**[Added to the original trac issue at 11.02am, Saturday, 31st January 2009]**

Wait before uploading.

I found the bug and will commit asap.
There was no data loss, so after the patch, your original file will work fine.

For the "unsaved changes", it actually means "not uploaded changes"... I'll change that

**Comment by tomhughes** [comment]
**Author: koying**
**[Added to the original trac issue at 11.19am, Saturday, 31st January 2009]**

(In [13452]) FIX : Bugs when loading MDC history (fixes #1546)
